Python/Product/Analysis/Pythia/AssignmentWalker.cs

```diff
@@ -0,0 +1,348 @@
+        /// <param name="m">The expression statement</param>
+        /// <param name="rightHandSide">Appended member calls</param>
+        private void HandleMemberExpression(string leftVariableName, MemberExpression m, string rightHandSide) {
+            var functionName = m.Name; // bitLength // count
+            if (m.Target is NameExpression n2) {
+                var functionInvokedOnName = n2.Name; // a 
+                if (functionInvokedOnName.Equals("self")) {
+                    return;
+                }
+
+                // if resolving an variable name based on previous definition
+                var resolvedName = Helper.ResolveVariable(Assignments, functionInvokedOnName);
+                functionInvokedOnName = !string.IsNullOrEmpty(resolvedName) ? resolvedName : functionInvokedOnName;
+
+                var combineName = functionInvokedOnName + "." + functionName + (!string.IsNullOrEmpty(rightHandSide) ? "." + rightHandSide : string.Empty);
+                Assignments.Add(new KeyValuePair() { Key = leftVariableName, Value = combineName });
+
+                return;
+            } else if (m.Target is CallExpression cc) {
+                if (cc.Target is MemberExpression mm) {
+                    HandleMemberExpression(leftVariableName, mm, (!string.IsNullOrEmpty(rightHandSide) ? functionName + "." + rightHandSide : functionName));
+                }
+            } else if (m.Target is MemberExpression mmm) {
+                HandleMemberExpression(leftVariableName, mmm, (!string.IsNullOrEmpty(rightHandSide) ? functionName + "." + rightHandSide : functionName));
+            }
+        }
+
+        /// <summary>
+        /// Determine the type of the variable and return a variable to type value pair
+        /// </summary>
+        /// <param name="variableName">Key</param>
+        /// <param name="value">Value to determine type for</param>
+        /// <returns>Key to value pair for typing</returns>
+        private KeyValuePair GetStandardVariableType(string variableName, string value) {
+            if (double.TryParse(value, out double val)) {
+                return new KeyValuePair() {
+                    Key = variableName,
+                    Value = StandardVariableTypes.Numeric
+                };
+            }
+
+            if (bool.TryParse(value, out bool val2)) {
+                return new KeyValuePair() {
+                    Key = variableName,
+                    Value = StandardVariableTypes.Boolean
+                };
+            }
+
+            return new KeyValuePair() {
+                Key = variableName,
+                Value = StandardVariableTypes.String
+            };
+        }
+    }
+
+    /// <summary>
+    /// Key value pair for variable name to type
+    /// </summary>
+    public class KeyValuePair {
+        public string Key { get; set; }
+        public string Value { get; set; }
+
+        public int SpanStart { get; set; }
```

Comment:
In most of the cases, `SpanStart` isn't assigned. Is that by design?

Reply:
No idea